### DISCUSSION TITLE
How to compute a symmetrical color wheel for the mapping of grains orientation

### DISCUSSION THREAD

**Discussion by AntoineMercier** [mtex-toolbox/mtex#1804] [Ask Anything] [answered]
Hello ! 

I have EBSD data from quartz measurements and I computed the crystallographic axes (c-axis and a-axis) orientations on a pole figure.
I would like to plot a map with the grains orientation of the quartz c-axis using a symmetrical color wheel (using the same color for 90° and 270° for example) in lower hemisphere as follows:
![Capture](https://user-images.githubusercontent.com/75066484/235186335-44e66264-ca6d-45cb-8bbc-06082355e195.PNG)

I looked through the forum and the documentation but I am a bit confused about how to compute that. 

Thank you for your help ! 

Best regards 

Antoine

**Comment by lfgmorales** [comment]
Hi

something like that?
```matlab
cdk=HSVDirectionKey(specimenSymmetry('-1'))
h = Miller({0,0,0,1},{1,1,-2,0},{1,0,-1,0},{1,0,-1,1},ebsd('q').CS)
cdir = ebsd('q').orientations.*h(1);
cdir.antipodal=1;
color = cdk.direction2color(cdir);
plot(ebsd('q'),color)
```
cheers

Luiz

On 28 Apr 2023, at 17:21, Antoine Mercier ***@***.******@***.***>> wrote:



Hello !

I have EBSD data from quartz measurements and I computed the crystallographic axes (c-axis and a-axis) orientations on a pole figure.
I would like to plot a map with the grains orientation of the quartz c-axis using a symmetrical color wheel (using the same color for 90° and 270° for example) in lower hemisphere as follows:
[Capture]<https://user-images.githubusercontent.com/75066484/235186335-44e66264-ca6d-45cb-8bbc-06082355e195.PNG>

I looked through the forum and the documentation but I am a bit confused about how to compute that.

Thank you for your help !

Best regards

Antoine

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1804>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN7JQMALQT6NP2BMF63XDPN7TANCNFSM6AAAAAAXPMCU4A>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1804", "url": "https://github.com/mtex-toolbox/mtex/discussions/1804", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by AntoineMercier** [reply]
Thank you very much for your quick answer! 

Yes this is almost what I am looking for: the map of c-axis orientation is perfect but the color wheel has still color ranging from 0  to 360 like that: 
![image](https://user-images.githubusercontent.com/75066484/235191397-3ddc9460-1fcb-4dcc-b3ed-cb9b739f664f.png)
 
I would like that the color range from 0 to 180: 
![image](https://user-images.githubusercontent.com/75066484/235192205-de8391e9-f8fd-4274-9c75-a73f61cd6ef5.png)

Is it possible to change the color wheel ?
thank you again

**Comment by ralfHielscher** [comment]
You can have a  threefold color wheel with
```matlab
cdk=HSVDirectionKey(crystalSymmetry('321'))
plot(cdk)
```
The question is whether this helps you :)

Ralf.





Ralf.

**Comment by kilir** [comment, marked as answer]
Hi,
I use this one here for mapping c-axes :+1: 
```
ck = HSVDirectionKey(specimenSymmetry('2'))
ck.whiteCenter=zvector;
ck.refl = zvector;
% ck.colorPostRotation=rotation.byAxisAngle(zvector,120*degree) % roate if you want

plot(ck)
```

There's a tiny glitch but so far I can live with it.

Cheers,
Rüdiger

**Reply by AntoineMercier** [reply]
This is precisely what I was looking for! Thanks a lot! 
Antoine
